Add subsections on "migration of the objects" (#57)
msagansk/Augur.Guide: collections/_version2-english/9-fork/4-migration-of-the-objects.md

{% capture image_src %}{{ "/" | absolute_url }}assets/images/{{page.collection}}/fork/migration-of-the-objects/forking-market.svg{% endcapture %}
{% include zoom-image.html src=image_src caption="Figure 3. migration of forking market"%}

# Migration of Non-Forking Markets (not finalized)
## Market Itself

> The non-forking markets which are not [finalized]({{glossary_finalized_market}}) can be migrated only to the [winning universe]({{glossary_winning_universe}}). To migrate those markets, you need to make a transaction for that explicitly. It can be done by anyone, not just the [market creator]({{glossary_market_creator}}), and whenever after finalizing the [forking market]({{glossary_forking_market}}). It is one-way and cannot be reversed.

Reviewer: ```suggestion
The non-forking markets which are not [finalized]({{glossary_finalized_market}}) can be migrated only to the [winning universe]({{glossary_winning_universe}}). To migrate those markets, you need to make a transaction for that explicitly. It can be done by anyone, not just the [market creator]({{glossary_market_creator}}), and anytime after finalizing the [forking market]({{glossary_forking_market}}). It is one-way and cannot be reversed.
```

> If you migrate the market which is past its [event end time]({glossary_end_time}) in the [parent universe]({{glossary_parent_universe}}), you need to supply [creation bond]({{glossary_creation_bond}}) (which is normally paid by the market creator when creating the market) and submit the [initial report]({{glossary_initial_report}}) in the [winning universe]({{glossary_winning_universe}}) for the market when you migrate it. Because all of the [REP]({{glossary_reputation_token}}) staked on the non-forking markets in the parent universe are refunded to their owner when a fork starts so that they migrate their REP to a [child universe]({{glossary_child_universe}}) during the [forking period]({{glossary_forking_period}}). Therefore, the creation bond is refunded to their owner when a fork starts, and the markets which are past their event end time are reset back to needing the initial report when migrated to a child universe.

Reviewer: ```suggestion
If you migrate a market that is past its [event end time]({glossary_end_time}) in the [parent universe]({{glossary_parent_universe}}), you need to supply [creation bond]({{glossary_creation_bond}}) (which is normally paid by the market creator when creating the market) and submit the [initial report]({{glossary_initial_report}}) in the [winning universe]({{glossary_winning_universe}}) for the market when you migrate it. Since all of the [REP]({{glossary_reputation_token}}) locked in the non-forking markets in the parent universe is refunded to its respective owners when a fork starts (so that they migrate their REP to a [child universe]({{glossary_child_universe}}) during the [forking period]({{glossary_forking_period}})), the first reporter must cover the first report REP costs themselves.
```

Reviewer: I tried like 6 times to get this paragraph to read smoother but didn't have much luck.  This is the best I could come up with, and it still isn't great.  😢

Author: Thanks for the improvement. And I'm feeling guilty for making you tried many times.

> I tried like 6 times to get this paragraph to read smoother but didn't have much luck. 

I believe it's because the original paragraph I wrote isn't good.
What prevents you from reading smoothly? My wording? Order of description? My poor vocabulary? For future reference, I would like to know.

> If you migrate the market which is past its [event end time]({glossary_end_time}) in the [parent universe]({{glossary_parent_universe}}), you need to supply [creation bond]({{glossary_creation_bond}}) (which is normally paid by the market creator when creating the market) and submit the [initial report]({{glossary_initial_report}}) in the [winning universe]({{glossary_winning_universe}}) for the market when you migrate it. Because all of the [REP]({{glossary_reputation_token}}) staked on the non-forking markets in the parent universe are refunded to their owner when a fork starts so that they migrate their REP to a [child universe]({{glossary_child_universe}}) during the [forking period]({{glossary_forking_period}}). Therefore, the creation bond is refunded to their owner when a fork starts, and the markets which are past their event end time are reset back to needing the initial report when migrated to a child universe.

Reviewer: You don't have to supply a new creation bond, but you do have to pay the REP for the first report.  Often these are the same size, but not always.

Author: I looked into the contracts:
https://github.com/AugurProject/augur/blob/2d47c7f1ee46f5e4b70e4ab2a267fd1f8bd6d067/packages/augur-core/src/contracts/reporting/Market.sol#L432-L445

To my untrained eye, it seems that:
 - Getting the amount of the REP bond in a child universe. (I believe the REP bond here is creation bond.)
 - The person who migrates a market has to supply the REP bond whose amount is gotten above👆. 
 - The person becomes the initial reporter.
 - If the market is past its event end time, then the person submits the initial report.
 
You said:
> You don't have to supply a new creation bond, but you do have to pay the REP for the first report.
 
But I think that anyone who migrates a market has to supply a new creation bond, though this bond is immediately used as the initial reporter's stake.

Reviewer: It does appear you are correct.  Comment withdrawn.  :)

Although the person who migrates a non-forking market needs to do somethings (such as supply creation bond, submit the initial report, and pay the [transaction fee]({{glossary_transaction_fee}}) for that), they can be a market creator of the market in the winning universe, so they will get [creator fees]({{glossary_creator_fee}}) paid by traders on the market in the winning universe. It might motivate migrating markets.

## Staked REP
When a [fork]({{glossary_fork}}) starts, all staked [REP]({{glossary_reputation_token}}) on the non-forking markets are un-staked and refunded to REP holders who staked it so that they migrate their REP during the [forking period]({{glossary_forking_period}}).

## Attached Objects
The objects which are attached to the non-forking market are:
 - [share]({{glossary_share}})
 - [unfilled order]({{glossary_unfilled_order}})
 - [creator fee]({{glossary_creator_fee}})
 - [validity bond]({{glossary_validity_bond}})

Those objects go where the market goes, they cannot be migrated separately.

> [Creation bond]({{glossary_creation_bond}}), which is paid by the [market creator]({{glossary_market_creator}}) when creating the market in the [parent universe]({{glossary_parent_universe}}), is refunded to either the market creator or the [initial reporter]({{glossary_initial_reporter}}). At the time of starting a [fork]({{glossary_fork}}), if the market has not gotten the [initial reporting]({{glossary_initial_report}}), this bond is refunded to the market creator. If already gets the initial reporting, refunded to the [initial reporter]({{glossary_initial_report}}). Because "gotten the initial reporting" means that the creation bond is already used by the initial reporter as the initial reporter's stake.

Reviewer: ```suggestion
[Creation bond]({{glossary_creation_bond}}), which is paid by the [market creator]({{glossary_market_creator}}) when creating the market in the [parent universe]({{glossary_parent_universe}}), is refunded to either the market creator or the [initial reporter]({{glossary_initial_reporter}}). At the time of starting a [fork]({{glossary_fork}}), if the market has not received an [initial reporting]({{glossary_initial_report}}), this bond is refunded to the market creator. If it already has received the initial report, then it refunds the [initial reporter]({{glossary_initial_report}}). Because "received the initial reporting" means that the creation bond is already used by the initial reporter as the initial reporter's stake.
```

# Migration of REP in Wallet

> All [REP]({{glossary_reputation_token}}) in the [parent universe]({{glossary_parent_universe}}) which are *not* staked on any [outcome]({{glossary_outcome}}) of [forking markets]({{glossary_forking_market}}) can be migrated to one of the [child universes]({{glossary_child_universe}}) by their owner. The migration of REP must be done during the [forking period]({{glossary_forking_period}}) which lasts 60 days after the [fork]({{glossary_fork}}) starts. After the forking period, the REP in the parent universe may not be migrated to any child universe permanently (See [to-do's]({{url_to_dos}}) for details).

Reviewer: ```suggestion
All [REP]({{glossary_reputation_token}}) in the [parent universe]({{glossary_parent_universe}}) that are *not* staked on any [outcome]({{glossary_outcome}}) of the [forking market]({{glossary_forking_market}}) can be migrated to one of the [child universes]({{glossary_child_universe}}) by their owner. The migration of REP must be done during the [forking period]({{glossary_forking_period}}) which lasts 60 days after the [fork]({{glossary_fork}}) starts. After the forking period, the REP in the parent universe can never be migrated to any child universe (See [to-do's]({{url_to_dos}}) for details).
```

> If you are the first person who migrates REP to a [child universe]({{glossary_child_universe}}), the child universe where you migrate your REP is created, and the forking market is also migrated to it. In this case, you have to pay the [transaction fee]({{glossary_transaction_fee}}) not only for your REP but also for creating the child universe and migrating the forking market.

Reviewer: ```suggestion
If you are the first person to migrate REP to a [child universe]({{glossary_child_universe}}), child universe creation is triggered and the forking market is also copied into it. In this case, you have to pay the [transaction fee]({{glossary_transaction_fee}}) not only for migrating your REP but also for creating the child universe and copying the forking market.
```
